### PR TITLE
Create module of IAM

### PR DESCRIPTION
**Pull request by mirik12**
- main.tf
- variables.tf
- output.tf
- README.md

### CODE REVIEW

File: terraform/modules/iam/main.tf
[the remark this answers is not in the document]
Comment (mirik12) [reply terraform/modules/iam/main.tf:1]: tfvars done

File: terraform/modules/iam/output.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+output "user_access_key_id" {
+  value       = aws_iam_access_key.user_key.id
+  description = "The access key ID for the IAM user."
+}
+
+output "user_secret_access_key" {
+  value       = aws_iam_access_key.user_key.secret
+  description = "The secret access key for the IAM user. This should be treated as sensitive information."
```
[the remark this answers is not in the document]
Comment (mirik12) [reply terraform/modules/iam/output.tf:13]: will add in next commit , thank you

File: terraform/modules/iam/main.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
+provider "aws" {
```
[the remark this answers is not in the document]
Comment (mirik12) [reply terraform/modules/iam/main.tf:1]: will add in next commit

File: terraform/env/dev/terraform.tfvars.example [hunk 3]
```diff
@@ -0,0 +1,3 @@
+aws_region            = "eu-central-1"
+aws_access_key_id     = ""
```
Comment (dereban25) [inline terraform/env/dev/terraform.tfvars.example:2]: don't use the access key and secret key. For local tests, you can use the profile via export AWS_PROFILE 
In CI/CD we will use this:
https://www.automat-it.com/post/using-github-actions-with-aws-iam-roles

File: terraform/modules/iam/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# IAM module for EKS
+
+## Example usage
+
+```terraform
+module "iam" {
+  source         = "../../modules/iam"
+  user_name      = "tf_user"
+  s3_bucket_name = "s3_name"
```
Comment (dereban25) [inline terraform/modules/iam/README.md:9]: everywhere using variables and values set in terraform.tfvars

File: terraform/modules/iam/main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (dereban25) [reply terraform/modules/iam/main.tf:1]: I think we will use the latest version. And will support the actual version.
Dont need to lock on providers and version

[the remark this answers is not in the document]
Comment (oleksiihead) [reply terraform/modules/iam/main.tf:1]: in any case, this step should be moved to the root module level

File: terraform/env/dev/terraform.tfvars.example (hunk 3 above)
Comment (Searge) [reply terraform/env/dev/terraform.tfvars.example:2]: > don't use the access key and secret key. For local tests, you can use the profile via export AWS_PROFILE In CI/CD we will use this: https://www.automat-it.com/post/using-github-actions-with-aws-iam-roles

@vanelin I said it to you :relieved:!

File: terraform/modules/iam/README.md (hunk 4 above)
Comment (Searge) [reply terraform/modules/iam/README.md:9]: Yeah. But that's just an example.
It is oblivious, that it should be in the variables.

File: terraform/modules/iam/output.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (mirik12) [reply terraform/modules/iam/output.tf]: we would not use our credentials , so removed it from output.tf in iam module

File: terraform/modules/iam/main.tf
[the remark this answers is not in the document]
Comment (mirik12) [reply terraform/modules/iam/main.tf:1]: at this moment we don't have any requirements for this .

**Pull request merged**